### ISSUE TITLE
Feature Request: Inject navigation in tree

### ISSUE THREAD

**Issue by dzannotti** [airbnb/native-navigation#56]
As i was trying out the library, i noticed that there's no way to have to inject the navigation anywhere within the react tree. I love the fact that each screen is atomic currently, but this would enable integration with flux libraries like (redux) and would allow to have persistent state across views (reinitialize redux every view isn't ideal).

A sub solution for this would be to be able to set an object that gets passed as props to each route, but i'd like much more a declarative api like
```
<Provider>
  <App>
     {(props) => <NativeNavigator {...props} />}
  </App>
</Provider>
```

This could possibly also indirectly fix the dependency on React mentioned here https://github.com/airbnb/native-navigation/issues/54#issuecomment-287887198

**Comment by Kureev** [comment]
> This could possibly also indirectly fix the dependency on React mentioned here #54 (comment)

I think there is a bit of misunderstanding here: React dependency mentioned in #54 is a native dependency. Current implementation assumes that React is a framework that Swift can import using `import React`. I think these issues are not related.

**Comment by lelandrichardson** [comment]
Indeed this issues are not related.

@dzannotti have you seen the [Integrating with Redux](http://airbnb.io/native-navigation/docs/guides/project-structure.html#integrating-with-redux) guide?

I have never felt any pain with adding redux or context-based top level elements to my app using native navigation's approach, so i'm not convinced that this being a real issue.

In your example, what component is rendering `<Provider>` here? Where is that component being mounted and rendered? What is it being rendered into?

Native navigation renders each screen into their own RCTRootView, and this may fundamentally go at odds with what you are proposing.

I hope that makes sense. Let me know if I'm not being clear.
